daml-lf/transaction/src/main/scala/com/daml/lf/TransactionNodeStatistic.scala

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package transaction
+
+import com.daml.lf.transaction.Transaction.ChildrenRecursion
+
+/** Container for transaction statistics.
+  *
+  * @param creates number of creates nodes,
+  * @param consumingExerciseByCids number of consuming exercises by contract ID nodes,
+  * @param nonconsumingExerciseByCids number of non-consuming Exercises by contract ID nodes,
+  * @param consumingExerciseByKeys number of consuming exercise by contract key nodes,
+  * @param nonconsumingExerciseByKeys number of non-consuming exercise by key nodes,
+  * @param fetcheByCids number of fetche nodes,
+  * @param fetchByKeys number of fetch by key  nodes,
+  * @param lookupByKeys number of lookup by key  nodes,
+  * @param rollbacks number of rollback nodes.
+  */
+final case class TransactionNodeStatistic(
+    creates: Int,
+    consumingExerciseByCids: Int,
+    nonconsumingExerciseByCids: Int,
+    consumingExerciseByKeys: Int,
+    nonconsumingExerciseByKeys: Int,
+    fetcheByCids: Int,
+    fetchByKeys: Int,
+    lookupByKeys: Int,
+    rollbacks: Int,
+) {
+
+  def +(that: TransactionNodeStatistic) =
+    TransactionNodeStatistic(
+      creates = this.creates + that.creates,
+      consumingExerciseByCids = this.consumingExerciseByCids + that.consumingExerciseByCids,
+      nonconsumingExerciseByCids =
+        this.nonconsumingExerciseByCids + that.nonconsumingExerciseByCids,
+      consumingExerciseByKeys = this.consumingExerciseByKeys + that.consumingExerciseByKeys,
+      nonconsumingExerciseByKeys =
+        this.nonconsumingExerciseByKeys + that.nonconsumingExerciseByKeys,
+      fetcheByCids = this.fetcheByCids + that.fetcheByCids,
+      fetchByKeys = this.fetchByKeys + that.fetchByKeys,
+      lookupByKeys = this.lookupByKeys + that.lookupByKeys,
+      rollbacks = this.rollbacks + that.rollbacks,
+    )
+
+  def exercisesByCid: Int = consumingExerciseByCids + nonconsumingExerciseByCids
+  def exercisesByKey: Int = consumingExerciseByKeys + nonconsumingExerciseByKeys
+  def exercises: Int = exercisesByCid + exercisesByKey
+  def fetches: Int = fetcheByCids + fetchByKeys
+  def actionNodes: Int = creates + exercises + fetches + lookupByKeys
```

Comment:
this is the only one where you add `Nodes` in the name. Maybe just call it `actions` or add nodes in the other names.

Reply:
:+1: